src/org/zaproxy/zap/extension/websocket/WebSocketProxy.java

```diff
@@ -1001,4 +1012,53 @@ public boolean isAllowAPI() {
 	public void setAllowAPI(boolean allowAPI) {
 		this.allowAPI = allowAPI;
 	}
+	
+	/**
+	 * Return an instance of {@link HandshakeConfig} which is used to establish a new WebSocket Connection
+	 * see {@link ServerConnectionEstablisher#send(HandshakeConfig)} and {@link WebSocketProxy#reEstablishConnection(HandshakeConfig)}
+	 * The {@link HandshakeConfig} includes the HttpHandshake of the current connection. In addition, the method adds the
+	 * {@link WebSocketObserver} and {@link WebSocketSenderListener}  of the current connection to HandshakeConfig
+	 *
+	 * @return the HandshakeConfig
+	 */
+	public HandshakeConfig getHandShakeConfig(boolean isFollowRedirection, boolean isStateRequest){
+		try {
+			HandshakeConfig handshakeConfig = new HandshakeConfig(new HttpMessage(handshakeReference.getHttpMessage().getRequestHeader()),isFollowRedirection,isStateRequest);
+			for(WebSocketObserver webSocketObserver : observerList){
+				handshakeConfig.addChannelObserver(webSocketObserver);
+			}
+			
+			for(WebSocketSenderListener senderListener : senderListenerList ){
+				handshakeConfig.addChannelSenderListener(senderListener);
+			}
+			return handshakeConfig;
+		} catch (HttpMalformedHeaderException e) {
+			e.printStackTrace();
```

Comment:
We should log the exceptions instead of printing. I'd suggest `logger.debug("Failed to read the HTTP handshake message:", e);`.